pydantic_ai_slim/pydantic_ai/builtin_tools.py

```diff
@@ -334,6 +336,55 @@ def unique_id(self) -> str:
         return ':'.join([self.kind, self.id])
 
 
+@dataclass(kw_only=True)
+class ToolSearchTool(AbstractBuiltinTool):
+    """A builtin tool that enables dynamic tool discovery without loading all definitions upfront.
+
+    Instead of consuming tokens on hundreds of tool definitions, Claude searches for relevant tools
+    on-demand. Only matching tools get expanded into full definitions in the model's context.
+
+    You should mark tools with `defer_loading=True` to make them discoverable on-demand while
+    keeping critical tools always loaded (with `defer_loading=False`).
+
+    Supported by:
+
+    * Anthropic (with `advanced-tool-use-2025-11-20` beta)
```

Comment:
We don't need to mention the beta, as we'd enable it automatically if this is used